pkg/token/federatedIdentity.go

```diff
@@ -27,8 +38,8 @@ func newWorkloadIdentityToken(clientID, federatedTokenFile, authorityHost, serve
 	if tenantID == "" {
 		return nil, errors.New("tenantID cannot be empty")
 	}
-	if federatedTokenFile == "" {
-		return nil, errors.New("federatedTokenFile cannot be empty")
+	if federatedTokenFile == "" && (os.Getenv(actionsIDTokenRequestToken) == "" || os.Getenv(actionsIDTokenRequestURL) == "") {
+		return nil, fmt.Errorf("either %s and %s environment variables have to be set or federated token file has to be provided", actionsIDTokenRequestToken, actionsIDTokenRequestURL)
```

Comment:
nit: this error can be extracted as a global value?

Reply:
i can change it to errors.New(), but what's the value of global var?

Reply:
i didn't opt to refactor this as there are many other errors.New() already